core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -1255,6 +1277,9 @@ private void writeValue(Object value) throws IOException {
             generator.writeEndObject();
         } else if (value instanceof Map) {
             writeMap((Map) value);
+        } else if (value instanceof Path) {
+            //Path implements Iterable<Path> and causes endless recursion and a StackOverFlow if treated as an Iterable here
+            generator.writeString(value.toString());            
```

Comment:
Checking out the `Path` JavaDoc, it does not appear that it will give the absolute path just by using `toString()`, but having thought about it, I agree that you do not want to make it absolute because that will change the path, which may only be relevant on the deserialization side. I'm imagining the creation of a `Path` on the client side, then passing it over the wire (having it serialized) where `../other/path/stuff` makes sense.